evalit/nifi/nifi_automation.py

```diff
@@ -364,44 +370,43 @@ def run_automation(self):
         )
         print("Status", r.status_code)
 
-        start_time_map = {}
-        end_time_map = {}
+        vals = self.parse_log(
+            log=log_file_location,
+            nfiles=len(self.files),
+            session_uuid=session_uuid,
+            poll_wait_time=5,
+        )
+        logger.debug(
+            f"Delta time for {self.__classname__} = {time.time() - start_automation}"
+        )
+        return vals
+
+    def parse_log(
+        self, log: str, nfiles: int, session_uuid: str, poll_wait_time: int = 5
+    ) -> Tuple[TransferDTO]:
+        timekeeper = {}
+        end_counter = 0
 
-        # Note from Nish:
-        # Need to avoid this infinite loop.
-        while 1:
-            # time.sleep(5)
+        # TODO: Optimize this!
+        while (len(timekeeper) < nfiles) and (end_counter < nfiles):
+            time.sleep(poll_wait_time)
             # read file line by line
-            with open(log_file_location, "r") as f:
+            with open(log, "r") as f:
                 for line in f:
                     if line.find(session_uuid) != -1:
```

Comment:
Could we use `.readlines()`?

Reply:
Oh ya. I just changed the log file name here. Will do that. (Previous code)